pennylane_cirq/simulator_device.py

```diff
@@ -81,26 +83,47 @@ def _apply_basis_state(self, basis_state_operation):
         if not self.shots is None:
             raise qml.DeviceError("The operation BasisState is only supported in analytic mode.")
 
-        basis_state_array = np.array(basis_state_operation.parameters[0])
+        wires = basis_state_operation.wires
 
-        if len(basis_state_array) != len(self.qubits):
+        if len(basis_state_operation.parameters[0]) != len(wires):
             raise qml.DeviceError(
                 "For BasisState, the state has to be specified for the correct number of qubits. Got a state for {} qubits, expected {}.".format(
-                    len(basis_state_array), len(self.qubits)
+                    len(basis_state_operation.parameters[0]), len(self.qubits)
                 )
             )
 
-        if not np.all(np.isin(basis_state_array, np.array([0, 1]))):
+        if not np.all(np.isin(basis_state_operation.parameters[0], np.array([0, 1]))):
             raise qml.DeviceError(
                 "Argument for BasisState can only contain 0 and 1. Got {}".format(
                     basis_state_operation.parameters[0]
                 )
             )
 
+        # expand basis state to device wires
+        basis_state_array = np.zeros(self.num_wires, dtype=int)
+        basis_state_array[wires] = basis_state_operation.parameters[0]
+
         self._initial_state = np.zeros(2 ** len(self.qubits), dtype=np.complex64)
         basis_state_idx = np.sum(2 ** np.argwhere(np.flip(basis_state_array) == 1))
```

Comment:
Oh, this is a permutation of converting from base2 to base10 that I hadn't seen before!

Interestingly, however, this seems to be a rare case where the native Python approach is still faster:

```pycon
>>> bs = np.array([1, 1, 0, 1, 1, 0, 0, 1, 1])
>>> int("".join(str(i) for i in bs), 2)
435
>>> np.sum(2 ** np.argwhere(np.flip(bs) == 1))
435
>>> %timeit np.sum(2 ** np.argwhere(np.flip(bs) == 1))
41.4 µs ± 4.82 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
>>> %timeit int("".join(str(i) for i in bs), 2)
14.4 µs ± 489 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```